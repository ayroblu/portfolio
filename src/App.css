.App {
  text-align: center;
}
.App.docked {
  padding-left: 200px;
}
.Drawer-top {
  padding: 10px 0;
  display: flex;
  align-items: center;
  justify-content: center;
  background: #D95B43;
  overflow: auto;
}
.Drawer-top > span{
  margin-left: 10px;
  color: white;
  font-size: 20px;
}

.App-header {
  background-color: #542437;
  height: 50px;
  padding: 20px;
  color: white;
}

.App-content {
  overflow: auto;
}
.App-text {
  font-size: large;
  padding: 0 10px;
}

.App-markdown {
  text-align: left;
  margin: 0 auto;
  padding: 0 20px;
  max-width: 700px;
}
.Languages {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  background-color: white;
  padding-bottom: 10px;
  margin: 10px 0;
}
.Languages > *{
  margin: 10px 10px 0;
}
.npm-logo{
  width: 200px;
}
.medium-logo{
  width: 300px;
}
.mathjax-node-parent{
  animation: fadeIn 2s 1s backwards ease-in;
}
@keyframes fadeIn{
  from{
    opacity: 0;
    filter: blur(10px);
  }
  80%{
    opacity: 0.5;
    filter: blur(5px);
  }
  to{
    opacity: 1;
    filter: none;
  }
}
