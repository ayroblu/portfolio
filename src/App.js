import React, { Component } from 'react'
import ReactMarkdown from 'react-markdown'
import * as MathJax from 'react-mathjax-updated'
import GoMarkGithub from 'react-icons/lib/go/mark-github'
import Drawer from 'material-ui/Drawer'
import MenuItem from 'material-ui/MenuItem'

import {cn} from './utils'
import mediumLogo from './img/medium-lockup-dark.png'
import npmLogo from './img/npm.png'
//import * as MathJax from 'react-mathjax'
import './App.css'

const vTerminal = `v_{terminal} = \\sqrt{\\frac{2 m g}{C_d A \\rho}}`
const vInitial = `v_{initial} = v_{arrow} \\times \\sin(\\theta)`
const uInitial = `v_{initial} = v_{arrow} \\times \\cos(\\theta)`
const v = `v = v_{terminal} \\times 
  \\frac{v_{initial} - v_{terminal} \\tan\\left(\\frac{t g}{v_{terminal}}\\right)}{
  v_{terminal} + v_{initial} \\tan\\left(\\frac{tg}{v_{terminal}}\\right)}`
const u = `u = \\frac{v_{terminal}^2 u_{initial}}{v_{terminal}^2 + g u_{initial} t}`
const velocity = `v_{abs} = \\sqrt{u^2 + v^2}`
const y = `y = \\frac{v_{terminal}^2}{2g} \\times \\log{\\left(\\frac{v_{initial}^2+v_{terminal}^2}{v^2 + v_{terminal}^2}\\right)}`
const x = `x = \\frac{v_{terminal}^2}{g} \\times \\log{\\left(\\frac{v_{terminal}^2 + g u_{initial} t}{v_{terminal}^2}\\right)}`

const targetAngle = `\\theta_{target} = \\tan^{-1}{\\left(\\frac{s_{vertical}}{s_{horizontal}}\\right)}`
const alpha = `\\alpha = \\theta_{target} - \\tan^{-1}{\\left(\\frac{s_{jaw} \\cos\\theta_{target}}{s_{horizontal}}\\right)}`
const phi = `\\phi = \\theta - \\alpha`
const sightHeight = `s_{sight} = s_{jaw} - s_{arm} \\tan(\\phi)`

class App extends Component {
  componentWillMount(){
    this.state = {open: window.innerWidth > 600}
  }
  componentDidMount() {
    window.addEventListener("resize", this.updatePage)
  }
  componentWillUnmount() {
    window.removeEventListener("resize", this.updatePage)
  }
  updatePage = ()=>{
    const docked = window.innerWidth > 600
    if (docked) {
      this.setState({open: true})
      return
    }
    this.forceUpdate()
  }
  render() {
    const docked = window.innerWidth > 600
    return (
      <div className={cn("App", docked && 'docked')}>
        <Drawer open={this.state.open}
          docked={docked}
          onRequestChange={open=>this.setState({open})}
          width={200}
        >
          <div className='Drawer-top'>
            <GoMarkGithub size={30} color='white'/><span>Ben's Profile</span>
          </div>
          <a href='#software'>
            <MenuItem>Software Development</MenuItem>
          </a>
          <a href='#npm'>
            <MenuItem>npm packages</MenuItem>
          </a>
          <a href='#medium'>
            <MenuItem>Medium articles</MenuItem>
          </a>
          <a href='#projects'>
            <MenuItem>Projects</MenuItem>
          </a>
          <a href='#react-native'>
            <MenuItem>React Native</MenuItem>
          </a>
          <a href='#android'>
            <MenuItem>Old Android Projects</MenuItem>
          </a>
          <a href='#other-projects'>
            <MenuItem>Other Projects</MenuItem>
          </a>
          <a href='#university'>
            <MenuItem>University of Auckland</MenuItem>
          </a>
        </Drawer>
        <div className="App-header">
          <h1>Ben Lu Portfolio</h1>
        </div>
        <div className='App-content'>
          <div>
            <p className="App-text">
              Hi, I'm a tinkerer and developer -
            </p><GoMarkGithub size={30} />
          </div>
          <p className="App-text">
            Need to show some kind of navigation - drawer - just use the materia-ui
          </p>
          <p className="App-text">
            As a developer I focus primarily on javascript projects, but have experience in Java, Ruby(on Rails), python, php, C, C++, C#, Visual Basic
          </p>
          <p className="App-text">
            Grouped languages with frameworks - images - links
          </p>
          <div className="App-text Languages">
            <div>
              <div>JavaScript</div>
              <div>Stars: 4.5</div>
            </div>
            <div>
              <div>Java</div>
              <div>Stars: 3</div>
            </div>
          </div>
          <div className="App-text">
            <img className='npm-logo' src={npmLogo} alt='npm logo' />
            npm - logo + package names and descriptions
      * db-difftool
          * CLI tool
          * Makes keeping track of databases easier
          * https://github.com/ayroblu/db-difftool
      * redux-add
          * CLI tool for adding redux files
          * https://github.com/ayroblu/redux-add
      * react-native-redux-log-monitor
          * React native component for using with react-native for displaying redux without js debugging
          * https://github.com/ayroblu/react-native-redux-log-monitor
      * object-type-check
        * This adds a basic type check at runtime thus allowing for a simple type check of objects
        * https://github.com/ayroblu/object-type-check
        * Testing for types
          </div>
          <p className="App-text">
            <img className='medium-logo' src={mediumLogo} alt='medium logo' />
    * https://hackernoon.com/server-side-rendering-with-create-react-app-1faf5a9d1eff
      * Source Code: https://github.com/ayroblu/ssr-create-react-app
    * https://medium.com/@benlu/ssr-with-create-react-app-v2-1b8b520681d9
      * Source Code: https://github.com/ayroblu/ssr-create-react-app-v2
      * Demo Deployed: https://ssr-cra-v2.now.sh/
          </p>
          <p className="App-text">
            projects - aklbuses.nz, video-delay, object-type-check
          </p>
          <p className="App-text">
            React Native
          </p>
          <p className="App-text">
            Old Android Projects
          </p>
          <p className="App-text">
            other projects - royalgameofur.site, ayro.nz, archery.ayro.nz, webgl?, selenium test?
          </p>
          <p className="App-text">
            University of Auckland - Commerce, Economics? (just show some old notes?)
          </p>
          <p className="App-text">
            University of Auckland - Engineering - See 4th year project + video - maybe discuss geothermal?
          </p>
          <p className="App-text">
            University of Washington - Exchange - photos
          </p>
          <p className="App-text">
            Archery - to bottom, just some beautiful archery shots and math - show all three bows
          </p>
          <ReactMarkdown className='App-markdown' source={`
# Who am I
* Archery
  * I have a recurve, compound and horsebow
  * I've made a couple of apps to help with my Archery
  * Solve these equations for arrow flight
`} />
          <MathJax.Context>
            <div className='mathjax-node-parent'>
              <MathJax.Node>{vTerminal}</MathJax.Node>
              <MathJax.Node>{vInitial}</MathJax.Node>
              <MathJax.Node>{uInitial}</MathJax.Node>
              <MathJax.Node>{v}</MathJax.Node>
              <MathJax.Node>{u}</MathJax.Node>
              <MathJax.Node>{velocity}</MathJax.Node>
              <MathJax.Node>{y}</MathJax.Node>
              <MathJax.Node>{x}</MathJax.Node>

              <MathJax.Node>{targetAngle}</MathJax.Node>
              <MathJax.Node>{alpha}</MathJax.Node>
              <MathJax.Node>{phi}</MathJax.Node>
              <MathJax.Node>{sightHeight}</MathJax.Node>
            </div>
          </MathJax.Context>

          <ReactMarkdown className='App-markdown' source={`
* University of Auckland
  * Bachelor of Engineering (Honours) specialising in Engineering Science
    * Operations Research
    * Fluid Dynamics
    * Geothermal Engineering
    * Software Engineering
  * Bachelor of Commerce majoring in Economics
    * Economics
    * Finance

# What I do
* Javascript: This is my primarily language
* npm packages
    * db-difftool
        * CLI tool
        * Makes keeping track of databases easier
        * https://github.com/ayroblu/db-difftool
    * redux-add
        * CLI tool for adding redux files
        * https://github.com/ayroblu/redux-add
    * react-native-redux-log-monitor
        * React native component for using with react-native for displaying redux without js debugging
        * https://github.com/ayroblu/react-native-redux-log-monitor
    * object-type-check
      * This adds a basic type check at runtime thus allowing for a simple type check of objects
      * https://github.com/ayroblu/object-type-check
      * Testing for types
* github
  * https://github.com/ayroblu
* Medium posts
  * https://hackernoon.com/server-side-rendering-with-create-react-app-1faf5a9d1eff
    * Source Code: https://github.com/ayroblu/ssr-create-react-app
  * https://medium.com/@benlu/ssr-with-create-react-app-v2-1b8b520681d9
    * Source Code: https://github.com/ayroblu/ssr-create-react-app-v2
    * Demo Deployed: https://ssr-cra-v2.now.sh/
* List of projects
    * Android (2013)
        * NZ Weather and Traffic
          * https://play.google.com/store/apps/details?id=benl.student.datacol&hl=en
        * Archery Scorer
          * https://play.google.com/store/apps/details?id=benl.student.archeryscorer
    * React Native
        * ArcheryNotes
          * https://github.com/ayroblu/ArcheryNotes
          * JS Testing: Jest + snapshot testing
    * First website
      * http://ben-l.com
    * aklbuses.nz
      * Live Auckland Buses
      * Shows the locations of Auckland buses in real time to the nearest 30 seconds
      * https://github.com/ayroblu/aklbuses.nz
    * webrec.ayro.nz
      * Video Delay
      * Simply uses chrome's video support, delays video for archery coaching
      * https://github.com/ayroblu/video-delay
    * royalgameofur.site
      * Royal Game of Ur
      * An implementation of the Royal Game of Ur based on: https://www.youtube.com/watch?v=WZskjLq040I
      * https://github.com/ayroblu/royal-game-of-ur
    * ayro.nz
      * https://ayro.nz/
      * Contact page: for business enquiries
    * archery.ayro.nz
      * https://archery.ayro.nz/
      * JS Testing: Jest + equation solving
* Selenium screenshot testing - take a snapshot, if different (b64) then generate a difference html page?
    * You can literally run a test, if they're different, run a different test / function in a new browser window
        * Use selenium to upload this file to it and show the difference between two images
          `}/>
        </div>
      </div>
    );
  }
}

export default App
