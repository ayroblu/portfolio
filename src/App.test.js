import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import injectTapEventPlugin from 'react-tap-event-plugin'
injectTapEventPlugin()
//import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <MuiThemeProvider>
      <App />
    </MuiThemeProvider>
  , div)

  //const tree = renderer.create(
  //  <App />
  //).toJSON()
  //expect(tree).toMatchSnapshot()
})
