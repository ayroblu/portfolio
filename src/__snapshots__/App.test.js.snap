// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <h1>
      Ben Lu Portfolio
    </h1>
  </div>
  <p
    className="App-text"
  >
    Hi, I'm a tinkerer and developer
  </p>
  <p
    className="App-text"
  >
    As a developer I focus primarily on javascript projects, but have experience in Java, Ruby(on Rails), python, php, C, C++, C#, Visual Basic
  </p>
  <p
    className="App-text"
  >
    Here's a list of things I do
  </p>
  <div
    className="App-markdown"
  >
    <h1
      className={undefined}
      data-sourcepos={undefined}
    >
      Who am I
    </h1>
    <ul
      className={undefined}
      data-sourcepos={undefined}
    >
      <li>
        <p>
          Archery
        </p>
        <ul
          className={undefined}
          data-sourcepos={undefined}
        >
          <li>
            I have a recurve, compound and horsebow
          </li>
          <li>
            I've made a couple of apps to help with my Archery
          </li>
          <li>
            Solve these equations for arrow flight
            <ul
              className={undefined}
              data-sourcepos={undefined}
            >
              <li>
                Insert LaTeX equations here
              </li>
            </ul>
          </li>
        </ul>
      </li>
      <li>
        <p>
          University of Auckland
        </p>
        <ul
          className={undefined}
          data-sourcepos={undefined}
        >
          <li>
            Bachelor of Engineering (Honours) specialising in Engineering Science
            <ul
              className={undefined}
              data-sourcepos={undefined}
            >
              <li>
                Operations Research
              </li>
              <li>
                Fluid Dynamics
              </li>
              <li>
                Geothermal Engineering
              </li>
              <li>
                Software Engineering
              </li>
            </ul>
          </li>
          <li>
            Bachelor of Commerce majoring in Economics
            <ul
              className={undefined}
              data-sourcepos={undefined}
            >
              <li>
                Economics
              </li>
              <li>
                Finance
              </li>
            </ul>
          </li>
        </ul>
      </li>
    </ul>
    <h1
      className={undefined}
      data-sourcepos={undefined}
    >
      What I do
    </h1>
    <ul
      className={undefined}
      data-sourcepos={undefined}
    >
      <li>
        <p>
          Machine Learning
        </p>
      </li>
      <li>
        <p>
          Javascript: This is my primarily language
        </p>
      </li>
      <li>
        <p>
          npm packages
        </p>
        <ul
          className={undefined}
          data-sourcepos={undefined}
        >
          <li>
            db-difftool
            <ul
              className={undefined}
              data-sourcepos={undefined}
            >
              <li>
                CLI tool
              </li>
              <li>
                Makes keeping track of databases easier
              </li>
              <li>
                https://github.com/ayroblu/db-difftool
              </li>
            </ul>
          </li>
          <li>
            redux-add
            <ul
              className={undefined}
              data-sourcepos={undefined}
            >
              <li>
                CLI tool for adding redux files
              </li>
              <li>
                https://github.com/ayroblu/redux-add
              </li>
            </ul>
          </li>
          <li>
            react-native-redux-log-monitor
            <ul
              className={undefined}
              data-sourcepos={undefined}
            >
              <li>
                React native component for using with react-native for displaying redux without js debugging
              </li>
              <li>
                https://github.com/ayroblu/react-native-redux-log-monitor
              </li>
            </ul>
          </li>
          <li>
            object-type-check
            <ul
              className={undefined}
              data-sourcepos={undefined}
            >
              <li>
                This adds a basic type check at runtime thus allowing for a simple type check of objects
              </li>
              <li>
                https://github.com/ayroblu/object-type-check
              </li>
              <li>
                Testing for types
              </li>
            </ul>
          </li>
        </ul>
      </li>
      <li>
        <p>
          github
        </p>
        <ul
          className={undefined}
          data-sourcepos={undefined}
        >
          <li>
            https://github.com/ayroblu
          </li>
        </ul>
      </li>
      <li>
        <p>
          Medium posts
        </p>
        <ul
          className={undefined}
          data-sourcepos={undefined}
        >
          <li>
            https://hackernoon.com/server-side-rendering-with-create-react-app-1faf5a9d1eff
            <ul
              className={undefined}
              data-sourcepos={undefined}
            >
              <li>
                Source Code: https://github.com/ayroblu/ssr-create-react-app
              </li>
            </ul>
          </li>
          <li>
            https://medium.com/@benlu/ssr-with-create-react-app-v2-1b8b520681d9
            <ul
              className={undefined}
              data-sourcepos={undefined}
            >
              <li>
                Source Code: https://github.com/ayroblu/ssr-create-react-app-v2
              </li>
              <li>
                Demo Deployed: https://ssr-cra-v2.now.sh/
              </li>
            </ul>
          </li>
        </ul>
      </li>
      <li>
        <p>
          List of projects
        </p>
        <ul
          className={undefined}
          data-sourcepos={undefined}
        >
          <li>
            Android (2013)
            <ul
              className={undefined}
              data-sourcepos={undefined}
            >
              <li>
                NZ Weather and Traffic
                <ul
                  className={undefined}
                  data-sourcepos={undefined}
                >
                  <li>
                    https://play.google.com/store/apps/details?id=benl.student.datacol&hl=en
                  </li>
                </ul>
              </li>
              <li>
                Archery Scorer
                <ul
                  className={undefined}
                  data-sourcepos={undefined}
                >
                  <li>
                    https://play.google.com/store/apps/details?id=benl.student.archeryscorer
                  </li>
                </ul>
              </li>
            </ul>
          </li>
          <li>
            React Native
            <ul
              className={undefined}
              data-sourcepos={undefined}
            >
              <li>
                ArcheryNotes
                <ul
                  className={undefined}
                  data-sourcepos={undefined}
                >
                  <li>
                    https://github.com/ayroblu/ArcheryNotes
                  </li>
                  <li>
                    JS Testing: Jest + snapshot testing
                  </li>
                </ul>
              </li>
            </ul>
          </li>
          <li>
            First website
            http://ben-l.com
          </li>
          <li>
            aklbuses.nz
            <ul
              className={undefined}
              data-sourcepos={undefined}
            >
              <li>
                Live Auckland Buses
              </li>
              <li>
                Shows the locations of Auckland buses in real time to the nearest 30 seconds
              </li>
              <li>
                https://github.com/ayroblu/aklbuses.nz
              </li>
            </ul>
          </li>
          <li>
            webrec.ayro.nz
            <ul
              className={undefined}
              data-sourcepos={undefined}
            >
              <li>
                Video Delay
              </li>
              <li>
                Simply uses chrome's video support, delays video for archery coaching
              </li>
              <li>
                https://github.com/ayroblu/video-delay
              </li>
            </ul>
          </li>
          <li>
            royalgameofur.site
            <ul
              className={undefined}
              data-sourcepos={undefined}
            >
              <li>
                Royal Game of Ur
              </li>
              <li>
                An implementation of the Royal Game of Ur based on: https://www.youtube.com/watch?v=WZskjLq040I
              </li>
              <li>
                https://github.com/ayroblu/royal-game-of-ur
              </li>
            </ul>
          </li>
          <li>
            ayro.nz
            <ul
              className={undefined}
              data-sourcepos={undefined}
            >
              <li>
                https://ayro.nz/
              </li>
              <li>
                Contact page: for business enquiries
              </li>
            </ul>
          </li>
          <li>
            archery.ayro.nz
            <ul
              className={undefined}
              data-sourcepos={undefined}
            >
              <li>
                https://archery.ayro.nz/
              </li>
              <li>
                JS Testing: Jest + equation solving
              </li>
            </ul>
          </li>
        </ul>
      </li>
      <li>
        <p>
          Selenium screenshot testing - take a snapshot, if different (b64) then generate a difference html page?
        </p>
        <ul
          className={undefined}
          data-sourcepos={undefined}
        >
          <li>
            You can literally run a test, if they're different, run a different test / function in a new browser window
            <ul
              className={undefined}
              data-sourcepos={undefined}
            >
              <li>
                Use selenium to upload this file to it and show the difference between two images
              </li>
            </ul>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</div>
`;
